modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -47,93 +47,180 @@ namespace {{packageName}}.Client
-        /// <param name="pathParams">Path parameters.</param>
-        /// <param name="authSettings">Authentication settings.</param>
-        /// <returns>Object</returns>
-        public Object CallApi(
-            String path, RestSharp.Method method, Dictionary<String, String> queryParams, String postBody,
-            Dictionary<String, String> headerParams, Dictionary<String, String> formParams,
-            Dictionary<String, FileParameter> fileParams, Dictionary<String, String> pathParams, String[] authSettings)
-        {
-            var request = PrepareRequest(
-                path, method, queryParams, postBody, headerParams, formParams, fileParams, pathParams, authSettings);
-            return (Object)RestClient.Execute(request);
-        }
+            public void AddFormParam(String name, Object value)
+            {
+                AddParam(formParams, name, value);
+            }
 
-        /// <summary>
-        /// Makes the asynchronous HTTP request.
-        /// </summary>
-        /// <param name="path">URL path.</param>
-        /// <param name="method">HTTP method.</param>
-        /// <param name="queryParams">Query parameters.</param>
-        /// <param name="postBody">HTTP body (POST request).</param>
-        /// <param name="headerParams">Header parameters.</param>
-        /// <param name="formParams">Form parameters.</param>
-        /// <param name="fileParams">File parameters.</param>
-        /// <param name="pathParams">Path parameters.</param>
-        /// <param name="authSettings">Authentication settings.</param>
-        /// <returns>The Task instance.</returns>
-        public async System.Threading.Tasks.Task<Object> CallApiAsync(
-            String path, RestSharp.Method method, Dictionary<String, String> queryParams, String postBody,
-            Dictionary<String, String> headerParams, Dictionary<String, String> formParams,
-            Dictionary<String, FileParameter> fileParams, Dictionary<String, String> pathParams, String[] authSettings)
-        {
-            var request = PrepareRequest(
-                path, method, queryParams, postBody, headerParams, formParams, fileParams, pathParams, authSettings);
-            return (Object) await RestClient.ExecuteTaskAsync(request);
+            public void AddFileParam(String name, Stream stream)
+            {
+                AddParam(fileParams, name, stream);
+            }
+            
+            public void SetPostBody(object value)
+            {
+                postBody = apiClient.Serialize(value);
+            }
+
+            public void AddAuthSettings(params String[] authMethod)
+            {
+                authSettings.AddRange(authMethod);
+            }
+
+            private RestRequest CreateRequest()
+            {
+                var request = new RestRequest(path, method);
+
+                var queryParamsCopy = new Dictionary<String, String>(queryParams);
```

Comment:
These copies of `queryParams` and `headerParams` are made before `UpdateParamsForAuth()` edits them in place. This is solely for keeping the builder object valid for repeat requests or requests after more options are added. Naturally, for this application such defensiveness is probably overkill. If we ensure that each builder is treated as valid for a maximum of one call, the copies can be removed and `*ParamsCopy` replaced with `*Params`.